crates/lib3h/src/transport/memory_mock/transport_memory.rs

```diff
@@ -115,59 +137,75 @@ impl Transport for TransportMemory {
                 uri
             )));
         }
-        let mut server = maybe_server.unwrap().lock().unwrap();
-        // Connect to it
-        server.connect(&my_uri.as_str())?;
-        // Generate and store connectionId
+        // Generate and store a connectionId to act like other Transport types
         self.n_id += 1;
         let id = format!("mem_conn_{}_{}", self.own_id, self.n_id);
-        self.connection_map.insert(id.clone(), uri.clone());
+        self.outbound_connection_map.insert(id.clone(), uri.clone());
+        // Connect to it
+        let mut server = maybe_server.unwrap().lock().unwrap();
+        server.request_connect(my_uri, &id)?;
         Ok(id)
     }
 
     /// Notify other server on that connectionId that we are closing connection and
     /// locally clear that connectionId.
     fn close(&mut self, id: &ConnectionIdRef) -> TransportResult<()> {
+        trace!("TransportMemory[{}].close({})", self.own_id, id);
+        if self.maybe_my_uri.is_none() {
+            return Err(TransportError::new(
+                "Cannot close a connection before bounding".to_string(),
+            ));
+        }
+        let my_uri = self.maybe_my_uri.clone().unwrap();
         // Get the other node's uri on that connection
-        let maybe_url = self.connection_map.get(id);
-        if let None = maybe_url {
-            return Err(TransportError::new(format!(
-                "No known connection for connectionId {}",
-                id
-            )));
+        let maybe_other_uri = self.outbound_connection_map.get(id);
+        if let None = maybe_other_uri {
+            return Err(TransportError::new(format!("Unknown connectionId: {}", id)));
         }
-        let url = maybe_url.unwrap();
+        let other_uri = maybe_other_uri.unwrap();
         // Get other node's server
         let server_map = memory_server::MEMORY_SERVER_MAP.read().unwrap();
-        let maybe_server = server_map.get(url);
-        if let None = maybe_server {
+        let maybe_other_server = server_map.get(other_uri);
+        if let None = maybe_other_server {
             return Err(TransportError::new(format!(
                 "No Memory server at this url: {}",
-                url,
+                other_uri,
             )));
         }
-        let mut server = maybe_server.unwrap().lock().unwrap();
-        // Tell it to close connection
-        server.close(&id)?;
+        let mut other_server = maybe_other_server.unwrap().lock().unwrap();
```

Comment:
Using something like a
```rust
let mut other_server = maybe_other_server.unwrap_or({ ... }).lock().expect("Fail to lock...");
```
could be easier to read, what do you think?

Reply:
doesn't work in Rust like previous comment.